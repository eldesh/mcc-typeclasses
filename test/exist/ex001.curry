-- The "standard" example for existential types

data Key a = forall b . Key b (b -> a)

key (Key x f) = f x

main = print (map key [Key "abc" length, Key 17 (+ 25), Key '\096' ord])
