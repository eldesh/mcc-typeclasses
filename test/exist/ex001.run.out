[3,42,96]
